### PR TITLE
Add instructions on how to build dev environment

### PR DESCRIPTION
**Pull request by dulek**
**What this PR does / why we need it**:
This commit enhances the CI playbooks to allow deploying from `master` branch and adds documentation on how to use them to deploy a development environment.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: docs/developers-guide.md [hunk 1]
```diff
@@ -42,6 +42,60 @@ Choose the one you are familiar with and easy to customize. Config the cluster w
...
+# Check if LB got created
+$ openstack loadbalancer list
++--------------------------------------+--------------------------------------+----------------------------------+-------------+---------------------+------------------+----------+
+| id                                   | name                                 | project_id                       | vip_address | provisioning_status | operating_status | provider |
++--------------------------------------+--------------------------------------+----------------------------------+-------------+---------------------+------------------+----------+
+| 9873d6d7-8ff1-4b5e-a8f0-6bb61f4dd58f | kube_service_kubernetes_default_test | deca4a226df049a689992105a65fbb66 | 10.1.0.36   | ACTIVE              | ONLINE           | amphora  |
++--------------------------------------+--------------------------------------+----------------------------------+-------------+---------------------+------------------+----------+
+
+# Call the LB
+$ curl 172.24.5.93
```
Comment (jichenjc) [inline docs/developers-guide.md:96]: I pulled the PR and tested locally it works fine

can this be updated as my env is different 
```
+--------------------------------------+---------------------+------------------+--------------------------------------+--------------------------------------+----------------------------------+
| ID                                   | Floating IP Address | Fixed IP Address | Port                                 | Floating Network                     | Project                          |
+--------------------------------------+---------------------+------------------+--------------------------------------+--------------------------------------+----------------------------------+
| 15332df2-567d-411c-9ad0-6b28664a6fd9 | 172.24.5.101        | 10.1.0.20        | 36ecb62e-33c0-454d-bc08-1cbe6541874f | 74990c2b-fab0-4be1-9294-435a4b12f872 | 58b5f2a307cf4961b4d4918581c6fb38 |
+--------------------------------------+---------------------+------------------+--------------------------------------+--------------------------------------+----------------------------------+

```

File: tests/ci-csi-cinder-e2e.sh [hunk 2]
```diff
@@ -32,7 +32,6 @@ cd "${REPO_ROOT}" || exit 1
 # PULL_NUMBER and PULL_BASE_REF are Prow job environment variables
 PR_NUMBER="${PULL_NUMBER:-}"
 [[ -z $PR_NUMBER ]] && echo "PR_NUMBER is required" && exit 1
-PR_BRANCH="${PULL_BASE_REF:-master}"
```
Comment (jichenjc) [inline tests/ci-csi-cinder-e2e.sh:35]: I assume this used to be the branch that related to the submited PR
is it not necessary or you used other variable to replace?

Comment (jichenjc) [reply tests/ci-csi-cinder-e2e.sh:35]: um... I didn't check all code but from the naming convention I think we will do something like 
master + PR => new test code thing to be tested on ?

Comment (dulek) [reply tests/ci-csi-cinder-e2e.sh:35]: Honestly I have no idea, but it does seem this is unused anywhere.
```
mdulko:cloud-provider-openstack/ (dev-environment) $ grep -R "github_pr_branch" .
mdulko:cloud-provider-openstack/ (dev-environment) $ 
```

Comment (jichenjc) [reply tests/ci-csi-cinder-e2e.sh:35]: to make it safe, can you help change some code (e.g update a log in the code )
and after the CI run, we check the output and see whether log is really there to prove the PR is running on top of master, if it works then let's remove the log and get this PR merge? 

Thanks

Comment (dulek) [reply tests/ci-csi-cinder-e2e.sh:35]: @jichenjc: Is there a need? If you take a look at [build-log](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/cloud-provider-openstack/2472/openstack-cloud-controller-manager-e2e-test/1722951771107627008/build-log.txt) you can see the `ci-occm-e2e.sh` script running with `-x` and there's no `PR_BRANCH` line anymore, yet everything works. It clearly runs the new code.

Comment (jichenjc) [reply tests/ci-csi-cinder-e2e.sh:35]: ok, thanks for the tip ,I think that make sense to me

**Comment by kayrus** [comment]
@dulek I'll try this on my local VM and let you know if it works smoothly.

**Comment by kayrus** [comment]
@dulek works great
/lgtm

**Comment by dulek** [comment]
Honestly I wouldn't bother too much with backporting this, it's just a development script.

**Pull request merged**